internal/provider/resource_access_generic.go

```diff
@@ -140,27 +155,290 @@ func (a *genericJAASAccessResource) Configure(ctx context.Context, req resource.
+	// Check first if the client is configured
+	if resource.client == nil {
+		addClientNotConfiguredError(&resp.Diagnostics, resource.resourceLogName, "update")
+		return
+	}
+
+	// Note: We only need to read the targetID from either the plan or the state.
+	// If it changed, the resource should be replaced rather than updated.
+
+	// Read Terraform configuration from the state
+	state, targetTag := resource.info(ctx, req.State, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Read Terraform configuration from the plan
+	plan, _ := resource.info(ctx, req.Plan, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get a diff of the plan vs. state to know what relations to add/remove
+	modelAdd, modelRemove := diffModels(plan, state, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Create a list of tuples to add and tuples to remove
+	addTuples := modelToTuples(ctx, targetTag, modelAdd, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	removeTuples := modelToTuples(ctx, targetTag, modelRemove, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Add new relations
+	if len(addTuples) > 0 {
+		err := resource.client.Jaas.AddRelations(addTuples)
+		if err != nil {
+			resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to add access rules, got error: %s", err))
+			return
+		}
+	}
+
+	// TODO: Consider updating the state here to reflect the newly added tuples before removing tuples in case the next removal fails.
+	// Would require an intermediate state.
 
+	// Delete removed relations
+	if len(removeTuples) > 0 {
+		err := resource.client.Jaas.DeleteRelations(removeTuples)
+		if err != nil {
+			resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to remove access rules, got error: %s", err))
+			return
+		}
+	}
+
+	// Set the desired plan onto the Terraform state after all updates have taken place.
+	resp.Diagnostics.Append(resource.save(ctx, &resp.State, plan, targetTag)...)
+}
+
+func diffModels(plan, state genericJAASAccessModel, diag *diag.Diagnostics) (toAdd, toRemove genericJAASAccessModel) {
+	newUsers := diffSet(plan.Users, state.Users, diag)
+	newGroups := diffSet(plan.Groups, state.Groups, diag)
+	newServiceAccounts := diffSet(plan.ServiceAccounts, state.ServiceAccounts, diag)
+	toAdd.Users = newUsers
+	toAdd.Groups = newGroups
+	toAdd.ServiceAccounts = newServiceAccounts
+	toAdd.Access = plan.Access
+
+	removedUsers := diffSet(state.Users, plan.Users, diag)
+	removedGroups := diffSet(state.Groups, plan.Groups, diag)
+	removedServiceAccounts := diffSet(state.ServiceAccounts, plan.ServiceAccounts, diag)
+	toRemove.Users = removedUsers
+	toRemove.Groups = removedGroups
+	toRemove.ServiceAccounts = removedServiceAccounts
+	toRemove.Access = plan.Access
+
+	return
+}
+
+func diffSet(current, desired basetypes.SetValue, diag *diag.Diagnostics) basetypes.SetValue {
```

Comment:
nit: desired is a misleading arg name based on the logic: given the current elements, return elements not in the second set.

Reply:
Added a docstring and changed desired to target.